src/modules_manifest.ts

```diff
@@ -0,0 +1,32 @@
+export interface FileMap<T> { [fileName: string]: T; }
+
+/** A class that maintains the module dependency graph of output JS files. */
```

Comment:
E.g. what is moduleToFileName?  (Is module the path to the module, its contents, some identifier?)

What is referencedModules?  a map of module to modules, or file to modules